main/src/library/File.flix

```diff
@@ -495,7 +532,109 @@ namespace File {
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            let fileStream = newFileStream(javaFile);
+            let printWriter = newPrintWriter(fileStream as ##java.io.Writer);
+
+            Foldable.foldLeft((_, b) -> write(fileStream, Int8.toInt32(b)), (), data);
+            close(printWriter);
+
+            if (checkError(printWriter)) {
+                Err("An error occurred when writing to the file: " + f)
+            }
+            else {
+                Result.flatMap((exists) -> Ok(not exists), alreadyExists)
+            }
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Appends `data` to the given `f`.
+    ///
+    /// Returns `true` if the file `f` was created, and `false` if `data` was appended to an existing `f`.
+    ///
+    pub def appendBytes(f: String, data: f[Int8]): Result[Bool, String] & Impure with Foldable[f] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileOutputStream(##java.io.File, Bool) as newFileStream;
+            import new java.io.PrintWriter(##java.io.Writer) as newPrintWriter;
+            import java.io.FileOutputStream.write(Int32);
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            // 2nd parameter `true` is for append.
+            let fileStream = newFileStream(javaFile, true);
+            let printWriter = newPrintWriter(fileStream as ##java.io.Writer);
+
+            Foldable.foldLeft((_, b) -> write(fileStream, Int8.toInt32(b)), (), data);
+            close(printWriter);
+
+            if (checkError(printWriter)) {
+                Err("An error occurred when writing to the file: " + f)
+            }
+            else {
+                Result.flatMap((exists) -> Ok(not exists), alreadyExists)
+            }
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns `true` if the file `f` was created, and `false` if `f` was overwritten.
+    ///
+    pub def truncate(f: String): Result[Bool, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            let printWriter = newPrintWriter(javaFile);
+
```

Comment:
Looks like you're missing some implementation here?

Reply:
Is it not supposed to clear the content of the file? If I run this code, the file is empty afterwards. Maybe there is a better solution. Any ideas?

Reply:
My understanding is that `truncate` should cut off the end of a file after a specified length. For example, a file containing the text `Hello` truncated to 3 bytes becomes `Hel`.

We can summon @magnus-madsen to be sure that's the right idea.

Reply:
I think it should zero the file, i.e. the file will exist, but it is empty.

Reply:
(it is not the same as deleting the file and recreating it-- because that would change permissions and timestamps.)

Reply:
I think the current code looks ok, although some suggest to use FileWriter instead of print writer. But other than that it is fine.

Reply:
@mlutze Truncating to n chars or n bytes will often not be meaningful. For example, with utf8 you cannot easily do it.

Reply:
In that case, perhaps the function could be called something like `clear`? For me, `truncate` means that there's some part remaining.

Reply:
I see the point, but I think its okay to call it `truncate` for now. (I took that name from another API). 

We could then have a `truncateTo` later?

Reply:
I don't like it, but it's not a hill I'll die on :)